4.0/en/0x21-V13-API.md

```diff
@@ -56,6 +56,18 @@ Note: Due to issues with XXE attacks against DTDs, DTD validation should not be
 | **13.4.1** | Verify that a query allow list or a combination of depth limiting and amount limiting is used to prevent GraphQL or data layer expression Denial of Service (DoS) as a result of expensive, nested queries. For more advanced scenarios, query cost analysis should be used. |  | ✓ | ✓ | 770 |
 | **13.4.2** | Verify that GraphQL or other data layer authorization logic should be implemented at the business logic layer instead of the GraphQL layer. |  | ✓ | ✓ | 285 |
 
+## V13.5 WebSocket Security Requirements
+
+| # | Description | L1 | L2 | L3 | CWE |
+| :---: | :--- | :---: | :---:| :---: | :---: |
+| **13.5.1** | Verify that only authenticated users can open WebSocket connections. | ✓ | ✓ | ✓ | 306 |
```

Comment:
Will web sockets only ever be for authenticated uses? Could they not also be used for unauthenticated purposes as well?

Reply:
Good point. Even though [some resources](https://www.freecodecamp.org/news/how-to-secure-your-websocket-connections-d0be0996c556/#-4-authenticate-users-before-ws-connection-establishes) say that "establishing a websocket connection is a bit expensive anyway", ASVS should in my opinion also allow unauthenticated websockets. I modified the rule.

Note that as stated in [this paper](https://www.researchgate.net/publication/315858773_Slowloris_DoS_Countermeasure_over_WebSocket), websocket is vulnerable to slowloris (not closing connections). It might be worth it to add another rule for connection limiting, to investigate...